src/wayland_clipboard.rs

```diff
@@ -16,41 +16,77 @@ limitations under the License.
 
 use std::error::Error;
 use std::ffi::c_void;
+use std::sync::{Arc, Mutex};
 
 use smithay_clipboard::WaylandClipboard;
-use wayland_client::Display;
+use wayland_client::protocol::wl_seat;
 use wayland_client::sys::client::wl_display;
+use wayland_client::{Display, GlobalManager, NewProxy};
 
 use common::ClipboardProvider;
 
 /// Clipboard context for Wayland clipboards.
 pub struct WaylandClipboardContext {
+    seat_name: Arc<Mutex<String>>,
     clip: WaylandClipboard,
 }
 
 impl WaylandClipboardContext {
     /// Create a new clipboard context.
-    pub fn new(display: &Display) -> Self {
+    pub fn new(display: &Display, manager: &GlobalManager) -> Self {
+        let seat_name = Arc::new(Mutex::new(String::new()));
+        let seat_name_clone = seat_name.clone();
+        manager
+            .instantiate_range(2, 6, move |proxy: NewProxy<wl_seat::WlSeat>| {
+                proxy.implement_closure(
+                    move |event, _| {
+                        if let wl_seat::Event::Name { name } = event {
+                            *seat_name_clone.lock().unwrap() = name
+                        }
+                    },
+                    (),
+                )
+            })
+            .unwrap();
+
         WaylandClipboardContext {
+            seat_name,
             clip: WaylandClipboard::new_threaded(display),
         }
     }
 
     /// Create a new clipboard context from an external pointer.
-    pub unsafe fn new_from_external(display: *mut c_void) -> Self {
+    pub unsafe fn new_from_external(display: *mut c_void, manager: &GlobalManager) -> Self {
+        let seat_name = Arc::new(Mutex::new(String::new()));
+        let seat_name_clone = seat_name.clone();
+        manager
+            .instantiate_range(2, 6, move |proxy: NewProxy<wl_seat::WlSeat>| {
+                proxy.implement_closure(
+                    move |event, _| {
+                        if let wl_seat::Event::Name { name } = event {
+                            *seat_name_clone.lock().unwrap() = name
+                        }
+                    },
+                    (),
+                )
+            })
+            .unwrap();
```

Comment:
This code should probably be moved to another independent function, so it doesn't have to be called twice.

I think it might actually be beneficial to separate concerns a bit here and just accept a seat and then provide a method on sctk or something like that to get the default seat?

I don't know where I'd get a `GlobalManager` from and I'd assume not everyone using the external display will have access to one. But even if you do not have access to it, `seat0` might still work great.

Let me know what you think so, I'm not sure if a `default_seat` function or something like that should be offered in rust-clipboard or sctk (or somewhere else), I just think that it might not be good to do a lot of wayland magic in rust-clipboard.